src/parallel/StaticIrregDomainDecomposition.cpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file StaticIrregDomainDecompostion.cpp
+ * @author amartyads
+ * @date 21.02.24
+ */
+
+#include "StaticIrregDomainDecomposition.h"
+#include "Domain.h"
+#include "utils/Logger.h"
+#include <fstream>
+#include <sstream>
+
+StaticIrregDomainDecomposition::StaticIrregDomainDecomposition(Domain* domain) : 
+	StaticIrregDomainDecomposition(domain, MPI_COMM_WORLD, {std::vector<unsigned int>{},{},{}}) {}
+
+StaticIrregDomainDecomposition::StaticIrregDomainDecomposition(
+	Domain* domain, MPI_Comm comm, const std::array<std::vector<unsigned int>, DIMgeom>& subdomainWeights) : 
+		DomainDecomposition(comm, {(int)subdomainWeights[0].size(), (int)subdomainWeights[1].size(), (int)subdomainWeights[2].size()}), 
+		_subdomainWeights(subdomainWeights),
+		_boxMin{0},
+		_boxMax{0},
```

Comment:
Either you specify all three values or non and rely on default initialization rules. But the way you wrote it you specify entry 0 and default 1 and 2. 

But I'd still argue to initialize them at their declaration.